mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -223,12 +223,12 @@ class KMeans private (
 
     // Compute squared norms and cache them.
     val norms = data.map(Vectors.norm(_, 2.0))
```

Comment:
`norms` have only one child `zippedData` so all actions that rely on `norms` also rely on `zippedData` and in runAlgorithm(), `zippedData` have been used multiple times.